File: CSES_Problem_List/Introductory_Problems/Permutations.cpp
/*
###########################################
Source: CSES

Problem statement:
A permutation of integers 1,2,\ldots,n is called beautiful if there are no adjacent elements whose difference is 1.
Given n, construct a beautiful permutation if such a permutation exists.

Constraints:
1 <= n <= 10^6

Solution: PTD504
###########################################
*/

#include <cstring>
#include <iostream>
#include <vector>
#include <cmath>
using namespace std;
#define ll long long
 
int main(){
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	int n; cin>>n;
	if (n == 1){
		cout << 1;
		return 0;
	}
	if (n <= 3){
		cout << "NO SOLUTION";
		return 0;
	}
	int temp = n - 1;
	cout << temp << " ";
	temp -= 2;
	while(temp > 0){
		cout << temp << " ";
		temp -= 2;
	}
	cout << n << " ";
	n -= 2;
	while(n > 0){
		cout << n << " ";
		n -= 2;
	}
}

File: CSES_Problem_List/Introductory_Problems/permutations.py
def permutations(n: int):
    if n == 1:
        return [1]
    
    if n < 4:
        print("NO SOLUTION")
        return []
    
    permutation = []
    val = 2

    while val <= n:
        permutation.append(val)
        val += 2
    
    val = 1

    while val <= n:
        permutation.append(val)
        val += 2
    
    return permutation

def main():
    n = int(input())
    permutation = permutations(n)

    if permutation:
        print(" ".join(map(str, permutation)))

if __name__ == "__main__":
    main()